app/services/provider_authorisation.rb

```diff
@@ -129,23 +133,34 @@ def full_authorisation?(permission:, course:)
     # enforce user-level permissions
```

Comment:
I think it's worth breaking `full_authorisation?` down into multiple smaller methods with descriptive names. It's quite central to permissions so it'll be worth having some easy to digest code over something that's logically efficient.